sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Trigger/WebPubSubTriggerDispatcher.cs

```diff
@@ -163,6 +183,17 @@ await executor.Executor.TryExecuteAsync(new TriggeredFunctionData
                     }
                     catch (Exception ex)
                     {
+                        if (context is MqttConnectionContext mqttContext && requestType == RequestType.Connect)
+                        {
+                            var mqttProtocolVersion = ((MqttConnectEventRequest)eventRequest).Mqtt.ProtocolVersion;
+                            var errorResponse = mqttProtocolVersion switch
+                            {
+                                MqttProtocolVersion.V311 => new MqttConnectEventErrorResponse(MqttV500ConnectReasonCode.ServerUnavailable, ex.Message),
+                                MqttProtocolVersion.V500 => new MqttConnectEventErrorResponse(MqttV500ConnectReasonCode.ServerUnavailable, ex.Message),
+                                _ => throw new NotSupportedException()
```

Comment:
usually we pass in the not supported value mqttProtocolVersion into NotSupportedException